jetty-http2/http2-server/src/main/java/org/eclipse/jetty/http2/server/HttpTransportOverHTTP2.java

```diff
@@ -443,7 +445,7 @@ private void abort(Throwable failure)
 
         private Throwable sending(Callback callback, boolean commit)
         {
-            synchronized (this)
+            try (AutoLock ignored = _lock.lock())
```

Comment:
```suggestion
            try (AutoLock lock = _lock.lock())
```

Reply:
Leave it as `ignored`, it's not a variable that is used.

Many different kinds of tooling will complain if we leave it as `lock` (checkstyle, findbugs, and error_prone will all complain)

Reply:
It is not ignored.  It is used.    I'm not going to change our code for bad tooling.   Checkstyle has not complained about this before???  If it does, then raise bugs on the tools as they should not complain if the variable is an AutoCloseable.

Reply:
@joakime absolutely not. That lock is used! Calling it "ignored" makes it look like it is not used.

Reply:
The `AutoLock` instance is used, the variable named `lock` is not.

We have code like this ...

``` java
try(AutoLock lock = _lock.lock()) {
   // the variable 'lock' is not used in this block.
}
```

This isn't new.

We have literally over 100 places in the existing Jetty codebase that honors this language quirk as well. (in both try-with-resources and catch blocks)

We have chosen to use the industry standard mechanism of using "ignored" or "ignore" as the variable name to signal to the tooling.
Or we can use the `@SuppressWarnings("unused")` before each instance as well.

This is not unique to the IDE, it's present in TONS of tooling around Java.
Even turning on the lint features on javac presents the same warnings.

If this is suddenly an issue for you, then vote for the variable-less language features then. (the same features that exist in other languages like C#)

``` java
try(_lock.lock()) { // no variable
}

and 

try {
} catch(IOException) { // no variable
}
```